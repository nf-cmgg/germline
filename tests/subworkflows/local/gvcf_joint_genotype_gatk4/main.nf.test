nextflow_workflow {

    name "Test Workflow GVCF_JOINT_GENOTYPE_GATK4"
    script "subworkflows/local/gvcf_joint_genotype_gatk4/main.nf"
    workflow "GVCF_JOINT_GENOTYPE_GATK4"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "gvcf_joint_genotype_gatk4"
    tag "vcf_concat_bcftools"
    tag "input_split_bedtools"

    test("gvcf_joint_genotype_gatk4 - single_sample") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", caller:"haplotypecaller"],
                    file(params.gvcf1, checkIfExists:true),
                    file(params.gtbi1, checkIfExists:true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists:true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists:true)
                ])
                input[3] = Channel.value([
                    [id:"dict"],
                    file(params.dict, checkIfExists:true)
                ])
                input[4] = [[],[]]
                input[5] = [[],[]]
                input[6] = false
                input[7] = 2
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcfs.collect { [it[0], "${file(it[1]).name},variantsMD5:${path(it[1]).vcf.variantsMD5}", file(it[2]).name] }
                ).match() }
            )
        }

    }

    test("gvcf_joint_genotype_gatk4 - family") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143,NA24149", caller:"haplotypecaller"],
                    file(params.gvcf1, checkIfExists:true),
                    file(params.gtbi1, checkIfExists:true)
                ],[
                    [id:"NA24149", sample:"NA24149", family:"Ashkenazim", family_samples:"NA24143,NA24149", caller:"haplotypecaller"],
                    file(params.gvcf2, checkIfExists:true),
                    file(params.gtbi2, checkIfExists:true)
                ]
                )
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists:true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists:true)
                ])
                input[3] = Channel.value([
                    [id:"dict"],
                    file(params.dict, checkIfExists:true)
                ])
                input[4] = [[],[]]
                input[5] = [[],[]]
                input[6] = false
                input[7] = 2
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcfs.collect { [it[0], "${file(it[1]).name},variantsMD5:${path(it[1]).vcf.variantsMD5}", file(it[2]).name] }
                ).match() }
            )
        }

    }

    test("gvcf_joint_genotype_gatk4 - only_merge") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", caller:"haplotypecaller"],
                    file(params.gvcf1, checkIfExists:true),
                    file(params.gtbi1, checkIfExists:true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists:true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists:true)
                ])
                input[3] = Channel.value([
                    [id:"dict"],
                    file(params.dict, checkIfExists:true)
                ])
                input[4] = [[],[]]
                input[5] = [[],[]]
                input[6] = true
                input[7] = 2
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcfs
                ).match() }
            )
        }

    }

    test("gvcf_joint_genotype_gatk4 - single_sample + family") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143,NA24149", caller:"haplotypecaller"],
                    file(params.gvcf1, checkIfExists:true),
                    file(params.gtbi1, checkIfExists:true)
                ],[
                    [id:"NA24149", sample:"NA24149", family:"Ashkenazim", family_samples:"NA24143,NA24149", caller:"haplotypecaller"],
                    file(params.gvcf2, checkIfExists:true),
                    file(params.gtbi2, checkIfExists:true)
                ],[
                    [id:"NA24385", sample:"NA24385", family:"NA24385", family_samples:"NA24385", caller:"haplotypecaller"],
                    file(params.gvcf3, checkIfExists:true),
                    file(params.gtbi3, checkIfExists:true)
                ]
                )
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists:true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists:true)
                ])
                input[3] = Channel.value([
                    [id:"dict"],
                    file(params.dict, checkIfExists:true)
                ])
                input[4] = [[],[]]
                input[5] = [[],[]]
                input[6] = false
                input[7] = 2
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcfs.collect { [it[0], "${file(it[1]).name},variantsMD5:${path(it[1]).vcf.variantsMD5}", file(it[2]).name] }
                ).match() }
            )
        }

    }
}
