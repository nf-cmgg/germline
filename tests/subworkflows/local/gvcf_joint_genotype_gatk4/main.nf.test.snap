{
    "gvcf_joint_genotype_gatk4 - single_sample + family": {
        "content": [
            [
                [
                    {
                        "family": "Ashkenazim",
                        "family_samples": "NA24143,NA24149",
                        "caller": "haplotypecaller",
                        "id": "Ashkenazim"
                    },
                    "Ashkenazim.haplotypecaller.vcf.gz,variantsMD5:4dea305eb71decb122709e75af9c833f",
                    "Ashkenazim.haplotypecaller.vcf.gz.tbi"
                ],
                [
                    {
                        "family": "NA24385",
                        "family_samples": "NA24385",
                        "caller": "haplotypecaller",
                        "id": "NA24385"
                    },
                    "NA24385.haplotypecaller.vcf.gz,variantsMD5:4ffd515511f59e3561e3fb1b046d7675",
                    "NA24385.haplotypecaller.vcf.gz.tbi"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-05T13:05:41.216399607"
    },
    "gvcf_joint_genotype_gatk4 - single_sample": {
        "content": [
            [
                [
                    {
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "caller": "haplotypecaller",
                        "id": "Ashkenazim"
                    },
                    "Ashkenazim.haplotypecaller.vcf.gz,variantsMD5:4c6db9171912bcbbaefeec2a24968a",
                    "Ashkenazim.haplotypecaller.vcf.gz.tbi"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-05T13:03:11.507733028"
    },
    "gvcf_joint_genotype_gatk4 - only_merge": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-05T11:43:25.386070284"
    },
    "gvcf_joint_genotype_gatk4 - family": {
        "content": [
            [
                [
                    {
                        "family": "Ashkenazim",
                        "family_samples": "NA24143,NA24149",
                        "caller": "haplotypecaller",
                        "id": "Ashkenazim"
                    },
                    "Ashkenazim.haplotypecaller.vcf.gz,variantsMD5:4dea305eb71decb122709e75af9c833f",
                    "Ashkenazim.haplotypecaller.vcf.gz.tbi"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-05T13:03:57.301900285"
    }
}